pyrsia_node/src/block_chain/block_chain.rs

```diff
@@ -0,0 +1,153 @@
+/*
+   Copyright 2021 JFrog Ltd
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+
+use chrono::Utc;
+use log::{info, warn};
+use serde_json::error::Error;
+use sha2::{Digest, Sha256};
+
+use crate::block_chain::block::Block;
+
+pub struct BlockChain {
+    blocks: Vec<Block>,
+}
+
+
+const DIFFICULTY_PREFIX: &str = "00";
+
+impl BlockChain {
+    pub fn new() -> &'static mut Self {
+        &mut BlockChain {
```

Comment:
Typically the `new` should look like this

```suggestion
    pub fn new() -> Self {
        BlockChain {
```

When you instantiate something you _always_ expect the caller to take ownership.

The new `BlockChain` instance allocated on line 34 will be destroyed on line 37 so the caller can not have a reference to the memory (that the compiler is trying to return).

Reply:
I orginally had this but that gave me other issues. Will retry

Reply:
@chb0github The problem is that you are returning a reference to the struct you instantiated. The struct itself has to go away when it goes out of scope.  Return the struct, not a reference to it, and then ownership moves to the caller.

Reply:
Yeah, I spend hours this weekend aligning my mind to ownership. I fixed it. It's not as I want it to be, but it works.